bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/BoschHttpClient.java

```diff
@@ -0,0 +1,187 @@
+                logger.info("Pairing failed with response status {}.", contentResponse.getStatus());
+                return false;
+            }
+        } catch (InterruptedException | TimeoutException | CertificateEncodingException | KeyStoreException e) {
+            logger.warn("Pairing failed with exception {}", e.getMessage());
+            return false;
+        } catch (ExecutionException e) {
+            // javax.net.ssl.SSLHandshakeException: General SSLEngine problem
+            // => usually the pairing failed, because hardware button was not pressed.
+            logger.trace("Pairing failed - Details: {}", e.getMessage());
+            logger.warn("Pairing failed. Was the Bosch SHC button pressed?");
+            return false;
+        }
+    }
+
+    public String createPairingUrl() {
+        return String.format("https://%s:8443/smarthome/clients", this.ipAddress);
+    }
+
+    public String createUrl(String endpoint) {
+        return String.format("https://%s:8444/%s", this.ipAddress, endpoint);
+    }
+
+    public String createSmartHomeUrl(String endpoint) {
+        return this.createUrl(String.format("smarthome/%s", endpoint));
+    }
+
+    public String createServiceUrl(String serviceName, String deviceId) {
+        return this.createSmartHomeUrl(String.format("devices/%s/services/%s/state", deviceId, serviceName));
+    }
+
+    public Request createRequest(String url, HttpMethod method) {
+        return this.createRequest(url, method, null);
+    }
+
+    public Request createRequest(String url, HttpMethod method, @Nullable Object content) {
+        Request request = this.newRequest(url).method(method).header("Content-Type", "application/json");
+        if (content != null) {
+            String body = gson.toJson(content);
+            logger.trace("create request for {} and content {}", url, body);
+            request = request.content(new StringContentProvider(body));
+        } else {
+            logger.trace("create request for {}", url);
+        }
+
+        // Set default timeout
+        request.timeout(10, TimeUnit.SECONDS);
+
+        return request;
+    }
+
+    /**
+     * Sends a request and expects a response of the specified type.
+     * 
+     * @param request Request to send
+     * @param responseContentClass Type of expected response
+     * @throws ExecutionException
+     * @throws TimeoutException
+     * @throws InterruptedException
+     */
+    public <TContent> TContent sendRequest(Request request, Class<TContent> responseContentClass)
+            throws InterruptedException, TimeoutException, ExecutionException {
+        ContentResponse contentResponse = request.send();
+
+        logger.debug("BoschHttpClient: response complete: {} - return code: {}", contentResponse.getContentAsString(),
+                contentResponse.getStatus());
+
+        @Nullable
+        TContent content = gson.fromJson(contentResponse.getContentAsString(), responseContentClass);
```

Comment:
You need to handle the syntax exception that gson throws here.

Reply:
Handling it now the same way as an empty response.